### PR TITLE
Fix some remote machine process tests

### LINKED ISSUE #18324: Remote machine process tests failing in CI
```
     System.Diagnostics.Tests.ProcessTests.TestProcessOnRemoteMachineWindows [FAIL]
        System.InvalidOperationException : Couldn't connect to remote machine.
        ---- System.InvalidOperationException : Process performance counter is disabled, so the requested operation cannot be performed.
        Stack Trace:
           D:\j\workspace\windows_nt_re---37265eab\src\System.Diagnostics.Process\src\System\Diagnostics\ProcessManager.Windows.cs(521,0): at System.Diagnostics.NtProcessManager.GetProcessInfos(String machineName, Boolean isRemoteMachine)
           D:\j\workspace\windows_nt_re---37265eab\src\System.Diagnostics.Process\src\System\Diagnostics\ProcessManager.Windows.cs(265,0): at System.Diagnostics.NtProcessManager.GetProcessIds(String machineName, Boolean isRemoteMachine)
           D:\j\workspace\windows_nt_re---37265eab\src\System.Diagnostics.Process\src\System\Diagnostics\ProcessManager.Windows.cs(33,0): at System.Diagnostics.ProcessManager.IsProcessRunning(Int32 processId, String machineName)
           D:\j\workspace\windows_nt_re---37265eab\src\System.Diagnostics.Process\src\System\Diagnostics\Process.cs(1055,0): at System.Diagnostics.Process.GetProcessById(Int32 processId, String machineName)
           D:\j\workspace\windows_nt_re---37265eab\src\System.Diagnostics.Process\tests\ProcessTests.cs(868,0): at System.Diagnostics.Tests.ProcessTests.<GetTestProcess>d__76.MoveNext()
           D:\j\workspace\windows_nt_re---37265eab\src\System.Linq\src\System\Linq\Select.cs(133,0): at System.Linq.Enumerable.SelectEnumerableIterator`2.MoveNext()
           ----- Inner Stack Trace -----
           D:\j\workspace\windows_nt_re---37265eab\src\System.Diagnostics.Process\src\System\Diagnostics\ProcessManager.Windows.cs(550,0): at System.Diagnostics.NtProcessManager.GetProcessInfos(PerformanceCounterLib library)
           D:\j\workspace\windows_nt_re---37265eab\src\System.Diagnostics.Process\src\System\Diagnostics\ProcessManager.Windows.cs(511,0): at System.Diagnostics.NtProcessManager.GetProcessInfos(String machineName, Boolean isRemoteMachine)
     System.Diagnostics.Tests.ProcessTests.GetProcessesByName_RemoteMachineNameWindows_ReturnsExpected(machineName: \"machine\") [FAIL]
        System.InvalidOperationException : Couldn't connect to remote machine.
        ---- System.InvalidOperationException : Process performance counter is disabled, so the requested operation cannot be performed.
        Stack Trace:
           D:\j\workspace\windows_nt_re---37265eab\src\System.Diagnostics.Process\src\System\Diagnostics\ProcessManager.Windows.cs(521,0): at System.Diagnostics.NtProcessManager.GetProcessInfos(String machineName, Boolean isRemoteMachine)
           D:\j\workspace\windows_nt_re---37265eab\src\System.Diagnostics.Process\src\System\Diagnostics\Process.cs(1107,0): at System.Diagnostics.Process.GetProcesses(String machineName)
           D:\j\workspace\windows_nt_re---37265eab\src\System.Diagnostics.Process\src\System\Diagnostics\Process.Windows.cs(27,0): at System.Diagnostics.Process.GetProcessesByName(String processName, String machineName)
           D:\j\workspace\windows_nt_re---37265eab\src\System.Diagnostics.Process\tests\ProcessTests.cs(821,0): at System.Diagnostics.Tests.ProcessTests.GetProcessesByName_ProcessNameMachineName_ReturnsExpected(String machineName)
           ----- Inner Stack Trace -----
           D:\j\workspace\windows_nt_re---37265eab\src\System.Diagnostics.Process\src\System\Diagnostics\ProcessManager.Windows.cs(550,0): at System.Diagnostics.NtProcessManager.GetProcessInfos(PerformanceCounterLib library)
           D:\j\workspace\windows_nt_re---37265eab\src\System.Diagnostics.Process\src\System\Diagnostics\ProcessManager.Windows.cs(511,0): at System.Diagnostics.NtProcessManager.GetProcessInfos(String machineName, Boolean isRemoteMachine)
     System.Diagnostics.Tests.ProcessTests.GetProcessesByName_RemoteMachineNameWindows_ReturnsExpected(machineName: \"\\\\machine\") [FAIL]
        System.InvalidOperationException : Couldn't connect to remote machine.
        ---- System.InvalidOperationException : Process performance counter is disabled, so the requested operation cannot be performed.
        Stack Trace:
           D:\j\workspace\windows_nt_re---37265eab\src\System.Diagnostics.Process\src\System\Diagnostics\ProcessManager.Windows.cs(521,0): at System.Diagnostics.NtProcessManager.GetProcessInfos(String machineName, Boolean isRemoteMachine)
           D:\j\workspace\windows_nt_re---37265eab\src\System.Diagnostics.Process\src\System\Diagnostics\Process.cs(1107,0): at System.Diagnostics.Process.GetProcesses(String machineName)
           D:\j\workspace\windows_nt_re---37265eab\src\System.Diagnostics.Process\src\System\Diagnostics\Process.Windows.cs(27,0): at System.Diagnostics.Process.GetProcessesByName(String processName, String machineName)
           D:\j\workspace\windows_nt_re---37265eab\src\System.Diagnostics.Process\tests\ProcessTests.cs(821,0): at System.Diagnostics.Tests.ProcessTests.GetProcessesByName_ProcessNameMachineName_ReturnsExpected(String machineName)
           ----- Inner Stack Trace -----
           D:\j\workspace\windows_nt_re---37265eab\src\System.Diagnostics.Process\src\System\Diagnostics\ProcessManager.Windows.cs(550,0): at System.Diagnostics.NtProcessManager.GetProcessInfos(PerformanceCounterLib library)
           D:\j\workspace\windows_nt_re---37265eab\src\System.Diagnostics.Process\src\System\Diagnostics\ProcessManager.Windows.cs(511,0): at System.Diagnostics.NtProcessManager.GetProcessInfos(String machineName, Boolean isRemoteMachine)
```

### PR DESCRIPTION
**Pull request by hughbe**
Fixes #18324

### CODE REVIEW

File: src/System.Diagnostics.Process/tests/ProcessTests.cs [hunk 1]
```diff
@@ -884,13 +883,23 @@ private static bool ProcessPeformanceCounterEnabled()
...
+                // The computer may be running with a guest user with process performance counters
+                // disabled. If so, try connecting to a remote machine. If the error contains any
+                // indication that process performance counters are disabled, then we will know.
+                try
+                {
+                    Process.GetProcesses(Guid.NewGuid().ToString("N"));
+                }
+                catch (InvalidOperationException ex)
+                {
+                    return !ex.Message.Contains("Process performance counter is disabled");
```
Comment (danmosemsft) [inline src/System.Diagnostics.Process/tests/ProcessTests.cs]: This test will fail on machines where .NET is localized. .NEt core is not, but .NET Framework is.

File: src/System.Diagnostics.Process/tests/ProcessTests.cs [hunk 2]
```diff
@@ -824,8 +824,7 @@ public void GetProcessesByName_ProcessNameMachineName_ReturnsExpected(string mac
             Assert.All(processes, process => Assert.Equal(machineName, process.MachineName));
         }
 
-        [ActiveIssue(18324)]
-        [Theory]
+        [ConditionalTheory(nameof(ProcessPeformanceCounterEnabled))]
```
Comment (danmosemsft) [inline src/System.Diagnostics.Process/tests/ProcessTests.cs]: can you fix `peformance` [sic]

Comment (danmosemsft) [inline src/System.Diagnostics.Process/tests/ProcessTests.cs]: Or, you could keep the code as you'v ewritten it but replace `return !ex.Message.Contains("Process performance counter is disabled");` with `return false;`

**Comment by hughbe** [comment]
I think it's safer just to disable the test if an exception was thrown. A message is written to the console anyway from Xunit:
```cs
   System.Diagnostics.Tests.ProcessTests.TestProcessOnRemoteMachineWindows [SKIP]
        Condition(s) not met: \"ProcessPerformanceCounterEnabled\"
     System.Diagnostics.Tests.ProcessTests.GetProcessesByName_RemoteMachineNameWindows_ReturnsExpected [SKIP]
        Condition(s) not met: \"ProcessPerformanceCounterEnabled\"
```

**Comment by hughbe** [comment]
Test Innerloop CentOS7.1 Debug x64 Build and Test
Test Innerloop Ubuntu14.04 Debug x64 Build and Test
Test Innerloop Ubuntu14.04 Release x64 Build and Test

**Comment by danmosemsft** [comment]
Thanks @hughbe

**Pull request merged**